### PR TITLE
add flag feature for parameters (issue #8)

### PR DESCRIPTION
**Pull request by PifPof73**
Hi,
I reforked your project to get the ...Tokyo... files back and my wrong ...Rio... files deleted. 
I just made the change in PasDoc_Utils.pas.

I made the changes in PasDoc_Types.pas like suggested by @michaliskambi, but the project group still does not compile in RIO with same error message:

`[dcc32 Error] PasDoc_Types.pas(197): E2010 Incompatible types: 'TStringArray' and 'System.TArray<System.string>`

Another problem during compile of all projects:
The text in PasDoc_Main.pas line 120ff is to long: 
`[dcc32 Error] PasDoc_Main.pas(120): E2056 String literals may have at most 255 elements`

But I don't want to edit your texts and make unwished changes.

issue #8 should be resolved.

Greetings,
PifPof73

### CODE REVIEW

**Comment by michaliskambi** [comment]
1. As for missing files - you didn't have to refork, you can always revert commits in GIT (this creates new commits "reverting" the changes from old commits) or rewrite GIT history and "force-push" the branch (less recommended if you're not experienced in GIT).

    The usual, safe, way to fix a pull request is to just make more commits, and push them :)

    Anyway, the state of previous pull request is saved -- it is visible in PR https://github.com/pasdoc/pasdoc/pull/82/files in "Files" tab. I'm not sure is it possible to restore it in a normal way (it seems you deleted the previous repository, the PR #82 points now to "unknown repository"). But you can always copy-paste stuff from https://github.com/pasdoc/pasdoc/pull/82/files . So if you lost something, it is still in https://github.com/pasdoc/pasdoc/pull/82/files

2. As for too long string in `PasDoc_Main.pas`: I think you fixed this in https://github.com/pasdoc/pasdoc/pull/82/files , and I was OK with your fix there, you made the sentence simpler this way. 

    In general the way to fix such "too long string literal" problems is to just make an expression adding 2 strings, i.e. change `'foobar'` into `'foo' + 'bar'`. 

    But in this case, changing that sentence into simpler is also very fine: `Read both interface and implementation comments and concatenate them. If the interface comment is also present (repeated) at the beginning of the implementation comment, it will be ignored (to not repeat 2x the same text in the concatenated result)`

3. As for `TStringArray` change: If you cannot find another solution, I'm not so much against just adding a typecast like you did `TNameParts(SplitString(s, '.'))`.

    I would prefer another way of solving it, by looking at what `SplitString` returns in the latest Delphi and adapting the code such that `TNameParts` is the same. Looking at this link http://docwiki.embarcadero.com/Libraries/Rio/en/System.StrUtils.SplitString it seems it returns now TStringDynArray , so it's actually compatible with FPC. So this should work:

```
type
{$IF NOT DECLARED(TStringDynArray)}
  TStringDynArray = array of string;
{$IFEND}
  TStringArray = TStringDynArray;
  TNameParts = TStringArray;
```

    Basically, I encourage you to make PasDoc compatible with Rio :) Don't be discouraged because I had some comments about the previous pull request -- I *always* have some comments, to every change :) Contributions are welcome, and I would like to have Rio compatiblity.

**Comment by michaliskambi** [comment]
You can also just revert changes to `PasDoc_Types.pas` in this pull request, and then keep this pull request limited to `PasDoc_Utils.pas` and only the changes to handle "flag" feature.

And then you can tackle Rio compatibility in another pull request.

**Comment by PifPof73** [comment]
I made a new commit to make the project group compilable with RIO.
I deleted my first fork - sorry for this mess - it is my first pull request using github and if I am honest - I am quiet confused with this web - gui.
I miss some git features for reverting inside github and struggeling with the merge / revert on the git console.

There for I closed my pull request and redone the fork and my changes.
Now I have two commits:
One for fixing issue #8 and now a second one for compiling with RIO.
For the Datatype I have to add System.Types into the uses section.

No problems for your comments - they are always welcome.
And I am learning from your comments, too - which is a perfect wini-win if we can help each other.

Please let me know, if I can do something else different.

During Covid-19 shutdown I can invest some time into this type of coding, so I will review the other issues - maybe I find something else to dive into.
Enjoy all your weekend.
Cheers,
PifPof73

**Comment by michaliskambi** [comment]
Thank you! Everything looks good now. Merging :)

**Comment by michaliskambi** [comment]
I pushed a few fixes, referenced in this pull request:

- https://github.com/pasdoc/pasdoc/commit/dde974def7a733be53187a06afabc45730353130 - fix FPC compilation

- https://github.com/pasdoc/pasdoc/commit/6732b002d4b600c86b320207e71aacb0247a878a and https://github.com/pasdoc/pasdoc/commit/ad744c74f0d2583a4b4ccc3dbe11281092f020a3 - fix the logic of flag detection. Our automatic tests found 2 problems.

- I also added a testcase, automatically processed and checked (by `make tests` and by Jenkins, it runs `make tests` automatically after every commit). You can see how it's done in https://github.com/pasdoc/pasdoc/commit/79fdabb56ba98fc9a83d6514c2ec4f60f8e56a0c (adding a testcase) and https://github.com/pasdoc/pasdoc/commit/b10c5ed3e8c1cc6cda7977ba9549873e567e19e7 (adding automatically-generated testcase output).
